torch/csrc/autograd/python_variable.cpp

```diff
@@ -803,7 +803,11 @@ static PyObject* THPVariable_make_wrapper_subclass(
 
     c10::SymInt size_bytes;
     auto dtype_itemsize = static_cast<int64_t>(options.dtype().itemsize());
-    if (sym_strides.has_value()) {
+    auto override_storage_size = r.toSymIntOptional(14);
+
+    if (override_storage_size.has_value()) {
+      size_bytes = override_storage_size.value();
```

Comment:
I think we should still do the computation below to ensure that "storage_size >= size_bytes". It would be a completely invalid Tensor if it indexes values outside of its storage.

Reply:
my entire goal here is just to skip the computation of size_bytes, because it fails for sizes that include a NestedInt. I don't really have a preference for how we do it - do you have a suggestion?

before this PR, NT uses size_bytes=0 here. so I imagine that the current implementation isn't any worse than before, at least for NT? I think your previous suggestion was also to use `size_bytes=1`

Reply:
Why does it fail for NestedInt? You should be able to compute the size of the values storage (assuming it's stored contiguously) right?

Reply:
https://github.com/pytorch/pytorch/blob/ebb8905e0c9c21f80a2428f43c9478324d1b4282/aten/src/ATen/EmptyTensor.cpp#L116

Addition is not implemented for NestedInt (and I'm not really sure we can/should implement it for NestedInt)

Reply:
I think we could discuss that more but it is beyond this PR for sure!
I would argue that it would be so cool to be able to do something like:
```python
# Assume I want a new Tensor with the first two dimensions collapsed
nested_size = my_nested_tensor.size()
nested_size[1] = nested_size[0] * nested_size[1]
nested_size.pop(0)
my_new_nested_tensor = torch.zeros(nested_size)
```

Reply:
interesting idea! filed an issue here: https://github.com/pytorch/pytorch/issues/125940 - perhaps Joel or Jeffrey have an idea of how to do this